### DISCUSSION TITLE
Nuxi build & start, how to precise the host?

### DISCUSSION THREAD

**Discussion by vince83110** [nuxt/nuxt#18033] [Questions] [answered]
I use a Docker environment for my Nuxt 3 application, using `nuxi dev`, I can precise the hostname (https://github.com/nuxt/framework/blob/b5618e976b7bae36d823ed607e548a901713a253/packages/nuxi/src/commands/dev.ts#L26), and this way, set 0.0.0.0 as host to go out from the Docker localhost.

On Vite, I can use `--host` option too, but I am kind of lost with Nitro / node / Vite in Nuxt 3.

How can build my app and then expose the host for a Docker container? For now, my traefik can't connect on the port 3000.

**Comment by Diizzayy** [comment, marked as answer]
Remove any changes pertaining to the `host` within your scripts or nuxt config, Then you can simply add `ENV HOST=0.0.0.0` to your Dockerfile

Here is an example of my Dockerfile below ( for a production app )

```
# STAGE 1 - Install Node Modules
FROM node:16-alpine as installer

WORKDIR /app

COPY package.json yarn.lock* ./

RUN yarn install --frozen-lockfile

# STAGE 2 - Build App
FROM node:16-alpine

WORKDIR /app

COPY --from=installer /app/node_modules ./node_modules

# Copy source files, and possibily invalidate so we have to rebuild
COPY . .

RUN yarn build

ENV HOST=0.0.0.0
ENV PORT=8080 

EXPOSE 8080

CMD node .output/server/index.mjs
```

**Reply by vince83110** [reply]
I understand the point, it's smart, I will test it today, thanks in anycase for your answer

**Reply by Diizzayy** [reply]
No problem, be sure to mark your question as answered if this indeed answers your question.

**Reply by vince83110** [reply]
Ok your solution worked perfectly, I didn't expose the port since I use Traefik, but it's very nice and nice layers for caching ! 

Thanks a lot, resolved.

**Reply by Diizzayy** [reply]
Happy to help!
